tests/api2/test_audit_basic.py

```diff
@@ -35,147 +84,156 @@ def initialize_for_smb_tests(request):
+
+        # Confirm existence of config entries
+        for key in [k for k in AUDIT_DATASET_CONFIG]:
+            assert key in config, str(config)
+
+        # Confirm audit default config settings
+        assert config['retention'] == AUDIT_CONFIG.defaults['retention']
+        assert config['quota'] == AUDIT_CONFIG.defaults['quota']
+        assert config['reservation'] == AUDIT_CONFIG.defaults['reservation']
+        assert config['quota_fill_warning'] == AUDIT_CONFIG.defaults['quota_fill_warning']
+        assert config['quota_fill_critical'] == AUDIT_CONFIG.defaults['quota_fill_critical']
+        assert config['remote_logging_enabled'] is False
+        for key in ['used', 'used_by_snapshots', 'used_by_dataset', 'used_by_reservation', 'available']:
+            assert key in config['space'], str(config['space'])
+
+        for service in ['MIDDLEWARE', 'SMB', 'SUDO']:
+            assert service in config['enabled_services']
+
+        # Confirm audit dataset settings
+        for key in [k for k in AUDIT_DATASET_CONFIG if AUDIT_DATASET_CONFIG[k] == 'zfs']:
+            assert get_zfs('zfs', key, dataset) == config[key], f"config[{key}] = {config[key]}"
+
+    def test_audit_config_dataset_defaults(self):
+        """ Confirm Audit dataset uses Audit default settings """
+        ds_config = call('audit.get_audit_dataset')
+        assert ds_config['org.freenas:quota_warning'] == AUDIT_CONFIG.defaults['quota_fill_warning']
+        assert ds_config['org.freenas:quota_critical'] == AUDIT_CONFIG.defaults['quota_fill_critical']
+
+    def test_audit_config_updates(self):
+        """
+        This test validates that setting values has expected results.
+        """
+        new_config = call('audit.update', {'retention': 10})
+        assert new_config['retention'] == 10
+
+        # quota are in units of GiB
+        new_config = call('audit.update', {'quota': 1})
+        assert new_config['quota'] == 1
+        audit_dataset = call('audit.get_audit_dataset')
+
+        # ZFS value is in units of bytes.  Convert to GiB for comparison.
+        assert get_zfs('zfs', 'refquota', audit_dataset) // GiB == new_config['quota']
+
+        # Confirm ZFS and audit config are in sync
+        assert new_config['space']['available'] == get_zfs('space', 'available', audit_dataset)
+        assert new_config['space']['used_by_dataset'] == get_zfs('space', 'used', audit_dataset)
+
+        # Check that we're actually setting the quota by evaluating available space
+        # Change the the quota to something more interesting
+        new_config = call('audit.update', {'quota': 2})
+        assert new_config['quota'] == 2
+
+        audit_dataset = call('audit.get_audit_dataset')
+        assert get_zfs('zfs', 'refquota', audit_dataset) == 2*GiB  # noqa (allow 2*GiB)
+
+        used_in_dataset = get_zfs('space', 'used_by_dataset', audit_dataset)
+        assert abs(new_config['space']['available'] - 2*GiB) == used_in_dataset  # noqa (allow 2*GiB)
```

Comment:
Subtraction seems backward (albeit inside an `abs`), but maybe add a comment to explain this test.

Reply:
I was re-using the original, but yeah, we can eliminate the `abs` with reversal and actually it makes for a better test since if the value goes negative that indicates the quota is not working!